File: Examples/Node.js/SDK/Charts/AddChart.js
// ExStart:1
var fs = require('fs');
var assert = require('assert');
var StorageApi = require('asposestoragecloud');
var CellsApi = require('asposecellscloud');
var configProps = require('../Config/config.json');
var data_path = '../../../../Data/';

var AppSID = configProps.app_sid;
var AppKey = configProps.api_key;
var config = {'appSid':AppSID,'apiKey':AppKey , 'debug' : true};

// Instantiate Aspose Storage API SDK
var storageApi = new StorageApi(config);
// Instantiate Aspose Cells API SDK
var cellsApi = new CellsApi(config);

// Set input file name
var name = "Sample_Test_Book.xls";
var sheetName = "Sheet5";
var chartType = "bar";
var upperLeftRow = 12;
var upperLeftColumn = 12;
var lowerRightRow = 20;
var lowerRightColumn = 20;
var area = "A1:A3";
var isVertical = false;
var isAutoGetSerialName = true;
var title = "SalesState";
try {
// Upload source file to aspose cloud storage
storageApi.PutCreate(name, null, null, data_path + name , function(responseMessage) {
	assert.equal(responseMessage.status, 'OK');

	// Invoke Aspose.Cells Cloud SDK API to add a chart in a worksheet
	cellsApi.PutWorksheetAddChart(name, sheetName, chartType, upperLeftRow,
			upperLeftColumn, lowerRightRow, lowerRightColumn, area,
			isVertical, null, isAutoGetSerialName,
			title, null, null, function(responseMessage) {

			assert.equal(responseMessage.status, 'OK');
			
			// Download updated Workbook from storage server
			storageApi.GetDownload(name, null, null, function(responseMessage) {
				assert.equal(responseMessage.status, 'OK');
				var outfilename = name;
				var writeStream = fs.createWriteStream(data_path + "_out_" + outfilename);
				writeStream.write(responseMessage.body);
				});
			});
	});

}catch (e) {
  console.log("exception in example");
  console.log(e);
}
//ExEnd:1

File: Examples/Android/app/src/main/java/com/aspose/cells/cloud/examples/chart/AddChart.java
package com.aspose.cells.cloud.examples.chart;

import android.content.Context;

import java.io.IOException;
import java.io.File;

import com.aspose.cells.api.CellsApi;
import com.aspose.cells.cloud.examples.Configuration;
import com.aspose.cells.cloud.examples.R;
import com.aspose.cells.cloud.examples.Utils;
import com.aspose.storage.api.StorageApi;

public class AddChart {

	public static void execute(Context context) throws IOException {
		// ExSart: convert-chart-to-image
		try {
			// Instantiate Aspose Storage API SDK
			StorageApi storageApi = new StorageApi(Configuration.apiKey, Configuration.appSID, true);

			// Instantiate Aspose Words API SDK
			CellsApi cellsApi = new CellsApi(Configuration.apiKey, Configuration.appSID, true);
			String input = "Sample1.xlsx";
			String output = "Sample2.xlsx";
			File inputFile = Utils.stream2file("sample1","xlsx", context.getResources().openRawResource(R.raw.sample1));
			File outputFile = Utils.stream2file("sample2","xlsx", context.getResources().openRawResource(R.raw.sample2));
			String sheet = "Sheet1";
			String chartType = "bar";
			Integer upperLeftRow = 12;
			Integer upperLeftColumn = 12;
			Integer lowerRightRow = 20;
			Integer lowerRightColumn = 20;
			String area = "A1:A3";
			Boolean isVertical = false;
			String categoryData = "";
			Boolean isAutoGetSerialName = true;
			String title = "SalesState";

			storageApi.PutCreate(input, null, Utils.STORAGE, inputFile);

			com.aspose.cells.model.ChartsResponse apiResposne = cellsApi.PutWorksheetAddChart(input, sheet, chartType,
					upperLeftRow, upperLeftColumn, lowerRightRow, lowerRightColumn, area, isVertical, categoryData,
					isAutoGetSerialName, title, Utils.STORAGE, Utils.FOLDER);

			com.aspose.storage.model.ResponseMessage sr = storageApi.GetDownload(input, null, Utils.STORAGE);

			Utils.copyInputStreamToFile(sr.getInputStream(),outputFile);

		}

		catch (Exception e) {
			e.printStackTrace();
		}
		// ExEnd: convert-chart-to-image

	}
}
